apps/examples/memory_fragmentation_test/memory_fragmentation_test.c

```diff
@@ -0,0 +1,318 @@
+/****************************************************************************
+ *
+ * Copyright 2019 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+/// @file memory_fragmentation_test.c
+
+/// @brief Intentionally allocate/free small and large memory segments in a mixed-up manner.
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+#include <stdlib.h>
+#include <time.h>
+#include <tinyara/mm/mm.h>
+
+/* The number of allocation memory size.
+ * Here, we have 12 different memory sizes of 2^4, 2^5, ... , 2^(MAX_SIZE_EXPONENT + 3)
+ */
+#define MAX_SIZE_EXPONENT 12
+
+/* We will consider the first NUMOF_SMALL_MEMORY sizes as relatively small memory
+ * That is, the sizes of 2^4 ~ 2^8 are considered as small memory.
+ */
+#define NUMOF_SMALL_MEMORY 5
+
+/* Seed for random number */
+#define SEED 1
+
+/* Data structure to store allocated memory segments */
+struct alloc_list {
+	char *data;
+	struct alloc_list *next;
+	struct alloc_list *prev;
+};
+
+static void memory_allocation(struct alloc_list list[], int numof_size[], int num_alloc[])
+{
+	struct alloc_list *next[MAX_SIZE_EXPONENT];
+	struct alloc_list *item;
+	int max_interval;
+	int i;
+	int interval[MAX_SIZE_EXPONENT] = {0, };
+	int cur_interval[MAX_SIZE_EXPONENT] = {0, };
+
+	max_interval = numof_size[0];
+	for (i = 0; i < MAX_SIZE_EXPONENT; ++i) {
+		next[i] = &list[i];
+		while (next[i]->next) {
+			next[i] = next[i]->next;
+		}
+	}
+
+	/* Calculate the maximum allocation count. */
+	for (i = 1; i < MAX_SIZE_EXPONENT; ++i) {
+		if (numof_size[i] > max_interval) {
```

Comment:
you can move this to above loop, if it is explicit code, you can ignore this comment.

Reply:
Done.